core/src/main/java/io/aiven/klaw/service/SchemaOverviewService.java

```diff
@@ -221,25 +229,50 @@ private SortedMap<Integer, Map<String, Object>> getSchemasMap(
       SortedMap<Integer, Map<String, Object>> schemaObjects)
       throws Exception {
     if (schemaUpdated) {
+
       for (MessageSchema messageSchema : topicSchemaVersionsInDb) {
         Map<String, Object> schemaObj = new HashMap<>();
-        schemaObj.put("schema", messageSchema.getSchemafull());
-        schemaObj.put("id", messageSchema.getSchemaId());
-        schemaObj.put("compatibility", messageSchema.getCompatibility());
+        schemaObj.put(SCHEMA, messageSchema.getSchemafull());
+        schemaObj.put(SCHEMA_ID, messageSchema.getSchemaId());
+        schemaObj.put(SCHEMA_COMPATIBILITY, messageSchema.getCompatibility());
         schemaObjects.put(Integer.parseInt(messageSchema.getSchemaversion()), schemaObj);
       }
     } else {
-      schemaObjects =
-          clusterApiService.getAvroSchema(
-              kwClusters.getBootstrapServers(),
-              kwClusters.getProtocol(),
-              kwClusters.getClusterName() + kwClusters.getClusterId(),
-              topicNameSearch,
-              tenantId);
+      schemaObjects = getSchemaFromAPI(topicNameSearch, tenantId, kwClusters);
+      if (schemaObjects != null) {
+        for (MessageSchema messageSchema : topicSchemaVersionsInDb) {
```

Comment:
can this be empty in this else block ?

Reply:
It should technically throw an exception if the api fails to retrieve the data but I feel like it is safer to keep it in.